app/service/file_svc.py

```diff
@@ -31,15 +31,25 @@ def __init__(self, payload_dirs, exfil_dir):
     async def upload(self, request):
         try:
             reader = await request.multipart()
-            field = await reader.next()
-            filename = field.filename
-            with open(os.path.join(self.exfil_dir, filename), 'wb') as f:
-                while True:
-                    chunk = await field.read_chunk()
-                    if not chunk:
-                        break
-                    f.write(chunk)
-            self.log.debug('Uploaded file %s' % filename)
+            exfil_dir = await self.create_unique_exfil_sub_directory()
+            while True:
+                field = await reader.next()
+                if not field:
+                    break
+                filename = field.filename
+                with open(os.path.join(exfil_dir, filename), 'wb') as f:
+                    while True:
+                        chunk = await field.read_chunk()
+                        if not chunk:
+                            break
+                        f.write(chunk)
+                self.log.debug('Uploaded file %s' % filename)
             return web.Response()
         except Exception as e:
             self.log.debug('Exception uploading file %s' % e)
+
+    async def create_unique_exfil_sub_directory(self):
```

Comment:
good idea. changes:

- make this a private function (_)
- put this function under a docstring called """ PRIVATE """ (look at the bottom of the operation_svc.py for an example of where the private functions are stored)